js/util/combineCriteria.ts

```diff
@@ -0,0 +1,31 @@
+import * as Collections from 'typescript-collections';
+import { PropExtractorCriteria, PropExtractorConfig } from './extractProps';
+
+export const getCombinedCriteria = (
+  componentName: string,
+  configs: PropExtractorConfig[]
+): PropExtractorCriteria => {
+  let remainingConfigs: PropExtractorConfig[] = [...configs];
+  let inclusionSet = new Collections.Set<string>();
+
+  while (remainingConfigs.length > 0) {
+    let config = remainingConfigs.shift();
+    let criteria = config[componentName];
+
+    if (!criteria) {
+      continue;
+    }
+
+    let configInclusionSet = new Collections.Set<string>();
+    let configExclusionSet = new Collections.Set<string>();
+    criteria.include.map(s => configInclusionSet.add(s));
```

Comment:
I don't think we have any specific guidance on this, but I generally prefer `forEach` when each element in an array is having some action performed with it (e.g. adding the element to a set), and `map` when each element in the array is being mapped to an element in a new array.

Reply:
Fair - fixed.